src/HPolyhedron.jl

```diff
@@ -484,11 +484,13 @@ function remove_redundant_constraints!(P::PT;
         br = b[non_redundant_indices]
         br[i] = b[j] + one(N)
         lp = linprog(-α, Ar, '<', br, -Inf, Inf, backend)
-        if lp.status != :Optimal
-            error("LP is not optimal")
+        if lp.status == :Infeasible
+            throw(DomainError(P, "the polyhedron is empty"))
+        else
+            error("LP is not optimal; the status of the LP is $(lp.status)")
         end
         objval = -lp.objval
-        if objval <= b[j]
+        if _leq(objval, b[j])
```

Comment:
I think we need to clarify the cases when we want to use `_leq` etc. Do we want to overapproximate or underapproximate the result? I think without an answer there is no point in using approximations.
The documentation should clearly mention the behavior for floating point, too.
In this case, I would say that we should not remove more constraints but rather less to be on the safe side, so here I would check for the opposite condition and use `!_geq(...)`.

Reply:
ok. for this change, i was having in mind https://github.com/JuliaReach/LazySets.jl/issues/754  (but i haven't tried it). then, i'll remove this commit which can be done in a separate PR.

i think that *once the floating point tolerance has been specified*, like a `ztol`, it doesn't matter if the result is under or overapproximated, one just wants that `objval` is either strictly smaller than `b[j]` or above but close to it. this is the purpose of `_leq` that is a relaxed version of `<=` for floating point. for rationals this is exact, and for FP we can pass the tolerances through an argument to  `remove_redundant_constraints!`.

Reply:
> i was having in mind #754

You relaxed the condition for removing, so you removed *more* constraints.

Reply:
ok. the examples there use `Polyhedra.removehredundancy!`, i should try in the first place how does it behave with `remove_redundant_constriants!`